drivers/tcs37727/tcs37727.c

```diff
@@ -0,0 +1,188 @@
+#define TCS37727_BDATAH            0x1B /* Blue ADC data high byte */
+#define TCS37727_PDATA             0x1C /* Proximity ADC data low byte */
+#define TCS37727_PDATAH            0x1D /* Proximity ADC data high byte */
+
+/**
+ * @brief Command Register
+ */
+#define TCS37727_BYTE_TRANS        0x80 /* Repeated byte protocol transaction */
+#define TCS37727_INC_TRANS         0xA0 /* Auto-increment protocol transaction */
+#define TCS37727_SF_PICLR          0xE5 /* Proximity interrupt clear */
+#define TCS37727_SF_CICLR          0xE6 /* Clear channel interrupt clear */
+#define TCS37727_SF_PCICLR         0xE7 /* Proximity and Clear channel interrupt clear */
+
+/**
+ * @brief Enable Register
+ */
+#define TCS37727_ENABLE_PIEN       (1 << 5) /* Proximity interrupt enable */
+#define TCS37727_ENABLE_AIEN       (1 << 4) /* Clear channel interrupt enable */
+#define TCS37727_ENABLE_WEN        (1 << 3) /* Wait enable, activates the wait feature */
+#define TCS37727_ENABLE_PEN        (1 << 2) /* Proximity enable, activates the proximity function */
+#define TCS37727_ENABLE_AEN        (1 << 1) /* RGBC enable, actives the two-channel ADC */
+#define TCS37727_ENABLE_PON        (1 << 0) /* Power ON */
+
+/**
+ * @brief Control Register
+ */
+#define TCS37727_CONTROL_PDRIVE_100    0x00 /* 100 mA LED Drive Strength */
+#define TCS37727_CONTROL_PDRIVE_50     0x04 /* 50 mA LED Drive Strength */
+#define TCS37727_CONTROL_PDRIVE_25     0x08 /* 25 mA LED Drive Strength */
+#define TCS37727_CONTROL_PDRIVE_12     0x0C /* 12.5 mA LED Drive Strength */
+#define TCS37727_CONTROL_AGAIN_1       0x00 /* 1 × gain RGBC Gain Value */
+#define TCS37727_CONTROL_AGAIN_4       0x01 /* 4 × gain RGBC Gain Value */
+#define TCS37727_CONTROL_AGAIN_16      0x02 /* 16 × gain RGBC Gain Value */
+#define TCS37727_CONTROL_AGAIN_60      0x03 /* 60 × gain RGBC Gain Value */
+
+/**
+ * @brief  Device ID
+ */
+#define TCS37727_DID_VALUE         0x49
+
+int tcs37727_test(tcs37727_t *dev)
+{
+    char id;
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_read_reg(dev->i2c, dev->addr, (TCS37727_BYTE_TRANS | TCS37727_ID), &id) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (id != TCS37727_DID_VALUE) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int tcs37727_init(tcs37727_t *dev, i2c_t i2c, uint8_t address)
+{
+    /* write device descriptor */
+    dev->i2c = i2c;
+    dev->addr = address;
+    dev->initialized = false;
+
+    i2c_acquire(dev->i2c);
+
+    /* initialize the I2C bus */
+    if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (tcs37727_test(dev)) {
+        return -2;
+    }
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_write_reg(dev->i2c, dev->addr,
+                      (TCS37727_BYTE_TRANS | TCS37727_CONTROL),
+                      TCS37727_CONTROL_AGAIN_4) != 1) {
```

Comment:
So you won't extend the functionality :-) ? Well ok, if this is merged I can can do it for you.